napari/layers/vectors/vectors.py

```diff
@@ -626,15 +626,29 @@ def _slice_data(
             while vectors passing through the current slice are assigned progressively lower
             values, based on how far from the current slice they originate.
         """
-        not_disp = list(self._dims_not_displayed)
-        indices = np.array(dims_indices)
+        # ensure dims not displayed is a list
+        dims_not_displayed = list(self._dims_not_displayed)
+
+        # We want a numpy array so we can use fancy indexing with the non-displayed
+        # indices, but as dims_indices can (and often/always does) contain slice
+        # objects, the array has dtype=object which is then very slow for the
+        # arithmetic below.
+        # promote slicing plane to arry so we can index into it, project as type float
+        not_disp_indices = np.array(dims_indices)[dims_not_displayed].astype(
+            float
+        )
```

Comment:
I guess all this can go inside the `if` clause right? Not a big deal though, so feel free to ignore.

Reply:
That's true. I've made the change.